// sanity-plugin-dashboard-widget-amplify/lib/amplify.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _default2 = _interopRequireDefault(require("part:@sanity/components/buttons/default"));

var _Amplify = _interopRequireDefault(require("./Amplify.css"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } }

function _createClass(Constructor, protoProps, staticProps) { if (protoProps) _defineProperties(Constructor.prototype, protoProps); if (staticProps) _defineProperties(Constructor, staticProps); return Constructor; }

function _possibleConstructorReturn(self, call) { if (call && (_typeof(call) === "object" || typeof call === "function")) { return call; } return _assertThisInitialized(self); }

function _getPrototypeOf(o) { _getPrototypeOf = Object.setPrototypeOf ? Object.getPrototypeOf : function _getPrototypeOf(o) { return o.__proto__ || Object.getPrototypeOf(o); }; return _getPrototypeOf(o); }

function _assertThisInitialized(self) { if (self === void 0) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function"); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, writable: true, configurable: true } }); if (superClass) _setPrototypeOf(subClass, superClass); }

function _setPrototypeOf(o, p) { _setPrototypeOf = Object.setPrototypeOf || function _setPrototypeOf(o, p) { o.__proto__ = p; return o; }; return _setPrototypeOf(o, p); }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var Amplify = /*#__PURE__*/function (_React$Component) {
  _inherits(Amplify, _React$Component);

  function Amplify() {
    var _getPrototypeOf2;

    var _this;

    _classCallCheck(this, Amplify);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _possibleConstructorReturn(this, (_getPrototypeOf2 = _getPrototypeOf(Amplify)).call.apply(_getPrototypeOf2, [this].concat(args)));

    _defineProperty(_assertThisInitialized(_this), "state", {
      error: null,
      deployLabel: null
    });

    _defineProperty(_assertThisInitialized(_this), "deploySite", function () {
      var deployLabel = "Deployment requested.";
      var webhook = _this.props.webhookUrl + "&operation=startbuild";

      _this.setState({
        deployLabel: deployLabel
      });

      fetch("https://webhooks.amplify.us-east-1.amazonaws.com/prod/webhooks?id=34b58cf5-f2d5-4dcb-af72-fcebae516e71&token=I1NRfbXU6uA9RbyQZK1QXkn0NryQn0Kzv90KchLMP38&operation=startbuild", {
        method: 'POST'
      }, {
        'Content-Type': 'application/json'
      });
    });

    return _this;
  }

  _createClass(Amplify, [{
    key: "render",
    value: function render() {
      var _this$state = this.state,
          deployLabel = _this$state.deployLabel,
          error = _this$state.error;
      var _this$props = this.props,
          title = _this$props.title,
          buttonText = _this$props.buttonText;

      if (error) {
        return _react["default"].createElement("pre", null, JSON.stringify(error, null, 2));
      }

      return _react["default"].createElement("div", {
        className: _Amplify["default"].container
      }, _react["default"].createElement("header", {
        className: _Amplify["default"].header
      }, _react["default"].createElement("h4", {
        className: _Amplify["default"].title
      }, title)), _react["default"].createElement("div", {
        className: _Amplify["default"].content
      }, _react["default"].createElement(_default2["default"], {
        className: _Amplify["default"].button,
        inverted: true,
        onClick: this.deploySite
      }, buttonText), _react["default"].createElement("span", {
        className: _Amplify["default"].status
      }, deployLabel)));
    }
  }]);

  return Amplify;
}(_react["default"].Component);

Amplify.propTypes = {
  title: _propTypes["default"].string,
  buttonText: _propTypes["default"].string,
  webhookUrl: _propTypes["default"].string
};
Amplify.defaultProps = {
  title: 'Deploy to Amplify',
  buttonText: "Deploy",
  webhookUrl: null
};
var _default = Amplify;
exports["default"] = _default;
